sql/delete.go

```diff
@@ -28,12 +29,20 @@ import (
 )
 
 // Delete deletes rows from a table.
+// Privileges: WRITE and READ on table. We currently always use a select statement.
+//   Notes: postgres requires DELETE. Also requires SELECT for "USING" and "WHERE" with tables.
+//          mysql requires DELETE. Also requires SELECT is a table is used in the "WHERE" clause.
 func (p *planner) Delete(n *parser.Delete) (planNode, error) {
 	tableDesc, err := p.getAliasedTableDesc(n.Table)
 	if err != nil {
 		return nil, err
 	}
 
+	if !tableDesc.HasPrivilege(p.user, parser.PrivilegeWrite) {
+		return nil, fmt.Errorf("user %s does not have %s privilege on table %s",
+			p.user, parser.PrivilegeWrite, tableDesc.Name)
```

Comment:
Eventually. My next step is to have real privileges. Once I have those sorted out, I'll probably just have a function that returns an error with the properly formatted message. For now, I'll leave it alone since I haven't quite determined what the final privilege handling will look like.